x-pack/plugins/infra/server/lib/alerting/common/utils.ts

```diff
@@ -107,3 +112,19 @@ export const getViewInAppUrlInventory = (
   const relativeViewInAppUrl = getInventoryViewInAppUrl(parseTechnicalFields(fields, true));
   return getViewInAppUrl(basePath, relativeViewInAppUrl);
 };
+
+export const getAlertUuidFromAlertId = async (esClient: ElasticsearchClient, alertId: string) => {
+  const { hits } = await esClient.search({
+    _source: false,
+    fields: [ALERT_UUID],
+    size: 1,
+    query: {
+      terms: {
+        [ALERT_RULE_UUID]: [alertId],
+      },
+    },
+  });
+  return hits.hits[0].fields?.[ALERT_UUID];
```

Comment:
Are we always fetching first alert from the rule? I am yet to test it locally, but looks like the alert UUID will be same for all alert notifications for a single rule.

Reply:
Now using the `executionId` instead, which should always lead to 1 result.

Reply:
After consulting with Chris, landed on a different approach which no longer needs an ES search